ubtue_SAGE Journals.js

```diff
@@ -0,0 +1,850 @@
+					},
+					{
+						"note": "orcid:0000-0003-2751-3204 | Halina Grzymała-Moszczyńska"
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://journals.sagepub.com/doi/10.1177/15423050211028189",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "Mitchell, Kenneth R., All Our Losses, All Our Griefs: Resources for Pastoral Care",
+				"creators": [
+					{
+						"lastName": "Johnson Brand",
+						"firstName": "Emi Alisa",
+						"creatorType": "author"
+					}
+				],
+				"date": "2021-09-01",
+				"DOI": "10.1177/15423050211028189",
+				"ISSN": "1542-3050",
+				"issue": "3",
+				"journalAbbreviation": "J Pastoral Care Counsel",
+				"language": "en",
+				"libraryCatalog": "utbue_SAGE Journals",
+				"pages": "229-230",
+				"publicationTitle": "Journal of Pastoral Care & Counseling",
+				"shortTitle": "Mitchell, Kenneth R., All Our Losses, All Our Griefs",
+				"url": "https://doi.org/10.1177/15423050211028189",
+				"volume": "75",
+				"attachments": [
+					{
+						"title": "SAGE PDF Full Text",
+						"mimeType": "application/pdf"
+					}
+				],
+				"tags": [
+					{
+						"tag": "Book Review"
+					}
+				],
+				"notes": [
+					{
+						"note": "<p>doi: 10.1177/15423050211028189</p>"
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://journals.sagepub.com/doi/full/10.1177/09518207221115929",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "The phallus in our stars: Sexual violence in the Animal Apocalypse",
+				"creators": [
+					{
+						"lastName": "Remington",
+						"firstName": "Megan R",
+						"creatorType": "author"
+					},
+					{
+						"lastName": "Smith",
+						"firstName": "Julianna Kaye",
+						"creatorType": "author"
+					}
+				],
+				"date": "2022-09-01",
+				"DOI": "10.1177/09518207221115929",
+				"ISSN": "0951-8207",
+				"issue": "1",
+				"language": "en",
+				"libraryCatalog": "utbue_SAGE Journals",
+				"pages": "57-74",
+				"publicationTitle": "Journal for the Study of the Pseudepigrapha",
+				"shortTitle": "The phallus in our stars",
+				"url": "https://doi.org/10.1177/09518207221115929",
+				"volume": "32",
+				"attachments": [
+					{
+						"title": "SAGE PDF Full Text",
+						"mimeType": "application/pdf"
+					}
+				],
+				"tags": [
+					{
+						"tag": "1 Enoch"
+					},
+					{
+						"tag": "animal symbolism"
+					},
+					{
+						"tag": "early Judaism"
+					},
+					{
+						"tag": "reception history"
+					},
+					{
+						"tag": "sexual violence"
+					}
+				],
+				"notes": [
+					{
+						"note": "<p>doi: 10.1177/09518207221115929</p>"
+					},
+					{
+						"note": "orcid:0000-0001-7826-6659 | Megan R Remington"
+					},
+					{
+						"note": "orcid:0000-0002-8956-2709 | Julianna Kaye Smith"
+					},
+					{
+						"note": "LF:"
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	}
+]
+/** END TEST CASES **/
```

Comment:
Not all cases from SAGE journals seem to be integrated:
```diff
2,3c2,3
<         "translatorID": "908c1ca2-59b6-4ad8-b026-709b7b927bda",
<         "label": "SAGE Journals",
---
>         "translatorID": "1f7dc3ef-761d-445c-8f49-efdf3d7bc978",
>         "label": "ubtue_SAGE Journals",
8c8
<         "priority": 100,
---
>         "priority": 99,
12c12
<         "lastUpdated": "2021-11-19 16:13:02"
---
>         "lastUpdated": "2022-11-07 17:31:18"
139c139,147
<                         
---
>                         //scrape ORCID from website e.g. https://doi.org/10.1177/09518207221115929
>                         let newAuthorSectionEntries = ZU.xpath(doc, '//span[@property="author"]');
>                         for (let authorSectionEntry of newAuthorSectionEntries) {
>                                 if (ZU.xpathText(authorSectionEntry, './a[contains(@href, "orcid")]')) {
>                                         let orcid = ZU.xpathText(authorSectionEntry, './a[contains(@href, "orcid")]').replace(/https?:\/\/orcid.org\//, '');
>                                         let authorName = ZU.xpathText(authorSectionEntry, './/span[@property="givenName"]') + ' ' + ZU.xpathText(authorSectionEntry, './/span[@property="familyName"]');
>                                         item.notes.push({"note": "orcid:" + orcid + ' | ' + authorName});
>                                 }
>                         }
151c159
<                                 }                               
---
>                                 }
153c161
<                          
---
> 
157,168c165
<                         // ubtue: extract translated and other abstracts from the different xpath
<                         var ubtueabstract = ZU.xpathText(doc, '//*[contains(concat( " ", @class, " " ), concat( " ", "abstractInFull", " " ))]');
<                         var otherabstract = ZU.xpathText(doc, '//article//div[contains(@class, "tabs-translated-abstract")]/p');
<                         var abstract = ZU.xpathText(doc, '//article//div[contains(@class, "abstractSection")]/p');
<                         if (abstract) {
<                                 item.abstractNote = abstract;
<                         }
<                         if (otherabstract) {
<                                 item.notes.push({note: "abs:" + otherabstract.replace(/^Résumé/, '')});
<                         } else {
<                                 item.abstractNote = ubtueabstract;
<                         }                       
---
>                         let absNr = 0;
169a167,171
>                         for (let abstract of ZU.xpath(doc, '//section[contains(@id,"abstract") and not(contains(@id,"abstracts"))]')) {
>                                         if (absNr == 0) item.abstractNote = abstract.textContent.replace(/^(Abstract|Résumé)/i, '');
>                                         else item.notes.push('abs:' + abstract.textContent.replace(/^(Abstract|Résumé)/i, ''));
>                                         absNr += 1
>                         }
184a187
>                         if (ZU.xpathText(doc, '//div[@class="meta-panel__type"]') && ZU.xpathText(doc, '//div[@class="meta-panel__type"]').match(/book\s+review/i)) item.tags.push("Book Review");
192c195
<                                                 item.abstractNote = ''; 
---
>                                                 item.abstractNote = '';
195c198,199
<                         }       
---
>                         }
> 
219c223
<                         
---
>                         else if (ZU.xpathText(doc, '//i[@class="icon-open_access"]/@data-original-title') == 'Open access') item.notes.push({note: 'LF:'});


```

Reply:
I removed these because I could not find them in any tested cases from SAGE and the whole SAGE-plattform was relounged a while ago. The heterogeneous approaches really complicate the treatment of abstracts.